src/renderer/before-frame-starts/runnables/setup-auto-registration.injectable.ts

```diff
@@ -0,0 +1,84 @@
+/**
+ * Copyright (c) OpenLens Authors. All rights reserved.
+ * Licensed under MIT License. See LICENSE in root directory for more information.
+ */
+import { getInjectable } from "@ogre-tools/injectable";
+import autoRegistrationEmitterInjectable from "../../../common/k8s-api/api-manager/auto-registration-emitter.injectable";
+import apiManagerInjectable from "../../../common/k8s-api/api-manager/manager.injectable";
+import { CustomResourceStore } from "../../../common/k8s-api/api-manager/resource.store";
+import type { CustomResourceDefinition } from "../../../common/k8s-api/endpoints";
+import { KubeApi } from "../../../common/k8s-api/kube-api";
+import { KubeObject } from "../../../common/k8s-api/kube-object";
+import type { KubeObjectStoreDependencies } from "../../../common/k8s-api/kube-object.store";
+import clusterFrameContextForNamespacedResourcesInjectable from "../../cluster-frame-context/for-namespaced-resources.injectable";
+import { evenBeforeClusterFrameStartsInjectionToken } from "../tokens";
+
+const setupAutoRegistrationInjectable = getInjectable({
+  id: "setup-auto-registration",
+  instantiate: (di) => ({
+    id: "setup-auto-registration",
+    run: () => {
+      const autoRegistrationEmitter = di.inject(autoRegistrationEmitterInjectable);
+      const beforeApiManagerInitializationCrds: CustomResourceDefinition[] = [];
+      const beforeApiManagerInitializationApis: KubeApi[] = [];
+      const deps: KubeObjectStoreDependencies = {
+        context: di.inject(clusterFrameContextForNamespacedResourcesInjectable),
+      };
+      let initialized = false;
+
+      const autoInitCustomResourceStore = (crd: CustomResourceDefinition) => {
+        const objectConstructor = class extends KubeObject {
+          static readonly kind = crd.getResourceKind();
+          static readonly namespaced = crd.isNamespaced();
+          static readonly apiBase = crd.getResourceApiBase();
+        };
+
+        const api = (() => {
+          const rawApi = apiManager.getApi(objectConstructor.apiBase);
+
+          if (rawApi) {
+            return rawApi;
+          }
+
+          const api = new KubeApi({ objectConstructor });
+
+          apiManager.registerApi(api);
+
+          return api;
+        })();
+
+        if (!apiManager.getStore(api)) {
+          apiManager.registerStore(new CustomResourceStore(deps, api));
+        }
+      };
+      const autoInitKubeApi = (api: KubeApi) => {
+        apiManager.registerApi(api);
+      };
+
+      autoRegistrationEmitter
```

Comment:
Thought: in a sense, `emitters` and their `listeners`, such as seen here, do something that `Runnable` could do, but without the burden of having to make a big deal about getting listeners "enlisted". That would be the same thing as `register`ing the `injectable`, which is more like a convention.

Reply:
Reminder: this is a hack to preserve behaviour while making sure circular dependencies don't exist.